packages/core/test/lib/console-child.js

```diff
@@ -0,0 +1,86 @@
+const assert = require("chai").assert;
+const fs = require("fs-extra");
+const path = require("path");
+const tmp = require("tmp");
+const TruffleConfig = require("@truffle/config");
+const { deriveConfigEnvironment } = require("../../lib/command-utils");
+
+let config;
+
+function createSandbox(source) {
+  if (!fs.existsSync(source)) {
+    throw new Error(`Sandbox failed: source: ${source} does not exist`);
+  }
+
+  const tempDir = tmp.dirSync({ unsafeCleanup: true });
+  fs.copySync(source, tempDir.name);
+  const config = TruffleConfig.load(
+    path.join(tempDir.name, "truffle-config.js"),
+    {}
+  );
+  return config;
+}
+
+describe("console-child", function () {
+  before("Create a sandbox", function () {
+    config = createSandbox(
+      path.join(__dirname, "..", "sources", "console-child")
+    );
+  });
+
+  describe("deriveConfigEnvironment", function () {
+    it("network has provider", function () {
+      config.network = "crazyTimeNetwork";
+      const expectedNetworkConfig = config.networks[config.network];
+      const cfg = deriveConfigEnvironment(config, config.network, "");
```

Comment:
Here you should probably just omit the third argument or explicitly pass `undefined` as I think that will most closely model the situation where `url` is not present. In other words, if no `url` is present, an empty string will not be passed but rather `undefined`.

Also a small stylistic nit; I would kind of rather specify the string literal `"crazyTimeNetwork"` as the second argument and not use `config.network`. I think it is slightly clearer and I'd rather be clearer than use another value that just happens to work.